### PR TITLE
Fix kafka consumer connection leak

### PR DESCRIPTION
**Pull request by jaime0815**
issue: #27222

### CODE REVIEW

**Comment by jaime0815** [comment]
pr: #27224

File: internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go [hunk 1]
```diff
@@ -6,16 +6,18 @@ import (
...
 	"github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper"
-	"go.uber.org/zap"
 )
 
 type Consumer struct {
 	c          *kafka.Consumer
 	config     *kafka.ConfigMap
 	msgChannel chan mqwrapper.Message
 	hasAssign  bool
+	hasChan    bool
```
Comment (xiaofan-luan) [inline internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go:20]: has to be an atomic bool

File: internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go [hunk 2]
```diff
@@ -226,9 +221,27 @@ func (kc *Consumer) GetLatestMsgID() (mqwrapper.MessageID, error) {
...
+	if cost > 200 {
+		log.Warn("close consumer costs too long time", zap.Any("topic", kc.topic), zap.String("groupID", kc.groupID), zap.Int64("time(ms)", cost))
+	}
+}
+
 func (kc *Consumer) Close() {
 	kc.closeOnce.Do(func() {
-		close(kc.closeCh)
-		kc.wg.Wait() // wait work goroutine exit, wait worker close the client
+		if kc.hasChan {
```
Comment (bigsheeper) [inline internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go:239]: What is the purpose of requiring a "hasChan"? How about:
```Go
kc.closeOnce.Do(func() {
    close(kc.closeCh)
    kc.closeInternal()
    kc.wg.Wait()
}
```

For non-blocking close:
```Go
kc.closeOnce.Do(func() {
    close(kc.closeCh)
    go kc.closeInternal()
}
```

**Pull request merged**